packages/cojson/src/sync.ts

```diff
@@ -818,6 +852,10 @@ export class SyncManager {
 
     const value = this.local.getCoValue(content.id);
 
+    if (value.isDeleted) {
+      storage.markCoValueAsDeleted(value.id);
+    }
```

Comment:
I wonder if this should happen in the same transaction that updates the content. Otherwise we risk ending up with inconsistent data in storage if `storage.markCoValueAsDeleted` completes and `storage.store` does not